From c6d3a45f40b55810f90749d9bad474f29aebcc68 Mon Sep 17 00:00:00 2001
From: Danesh M <daneshm90@gmail.com>
Date: Thu, 16 Apr 2015 10:18:08 -0700
Subject: [PATCH 31/74] FMRadio : Launch station list automatically after scan

Give status update post scaning for categories
Fix %s showing in stationList

Change-Id: I1845f0cdeb0051bcfb951e2fef791a016fa81ef4
---
 fmapp2/res/values/strings.xml           |  3 ++-
 fmapp2/src/com/caf/fmradio/FMRadio.java | 24 +++++++++++++++++++++++-
 2 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/fmapp2/res/values/strings.xml b/fmapp2/res/values/strings.xml
index 1076c09..991b634 100644
--- a/fmapp2/res/values/strings.xml
+++ b/fmapp2/res/values/strings.xml
@@ -125,7 +125,7 @@
     <string name="default_blank" translatable="false"></string>
     <string name="add_station" translatable="false">+</string>
 
-    <string name="station_name">Station: <xliff:g id="name">%s</xliff:g></string>
+    <string name="station_name">Station:</string>
     <string name="station_exists">This frequency already exists!</string>
     <string name="station_list_delete_station">Delete: <xliff:g id="name">%s</xliff:g></string>
     <string name="station_list_delete_station_prompt">Are you sure you want to delete \'<xliff:g id="name">%s</xliff:g>\'?</string>
@@ -141,4 +141,5 @@
     <string name="set">Set</string>
     <string name="user_defind_band_msg">Enter frequency from 76.0 - 108.0, with minimum 1 channel spacing and 100KHz space between max and min freq</string>
     <string name="save_record_file">Recorded FM file saved to "<xliff:g id="record_file">%1$s</xliff:g>"</string>
+    <string name="fm_search_no_results">No compatible stations found</string>
 </resources>
diff --git a/fmapp2/src/com/caf/fmradio/FMRadio.java b/fmapp2/src/com/caf/fmradio/FMRadio.java
index 635e47f..4215b80 100644
--- a/fmapp2/src/com/caf/fmradio/FMRadio.java
+++ b/fmapp2/src/com/caf/fmradio/FMRadio.java
@@ -236,7 +236,7 @@ public class FMRadio extends Activity
    // default audio device - speaker
    private static int mAudioRoute = FMRadioService.RADIO_AUDIO_DEVICE_WIRED_HEADSET;
    private static boolean mFMStats = false;
-
+   private boolean mShowStationList = false;
 
    /* Current Status Indicators */
    private static boolean mRecording = false;
@@ -1766,6 +1766,9 @@ public class FMRadio extends Activity
       SharedPreferences.Editor editor = sp.edit();
       editor.clear();
       editor.commit();
+      if (mScannedFrequencies != null) {
+         mScannedFrequencies.clear();
+      }
    }
    public boolean fmConfigure() {
       boolean bStatus = true;
@@ -2743,6 +2746,11 @@ public class FMRadio extends Activity
          resetFMStationInfoUI();
          invalidateOptionsMenu();
          saveStations();
+         if (mShowStationList) {
+            Intent stationListIntent = new Intent(FMRadio.this, StationListActivity.class);
+            startActivity(stationListIntent);
+         }
+         mShowStationList = false;
       }
    };
 
@@ -3188,6 +3196,20 @@ public class FMRadio extends Activity
       }
       public void onSearchComplete() {
          Log.d(LOGTAG, "mServiceCallbacks.onSearchComplete :");
+         if (mIsScaning) {
+             if (mScannedFrequencies != null && mScannedFrequencies.size() > 0) {
+                 mShowStationList = true;
+             } else {
+                 mHandler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        Toast t = Toast.makeText(FMRadio.this,
+                           getString(R.string.fm_search_no_results), Toast.LENGTH_SHORT);
+                        t.show();
+                    }
+                 });
+             }
+         }
          mScanPty = 0;
          mScanPtyIndex = 0;
          mIsScaning = false;
-- 
2.17.1

